extensions/smallrye-jwt/runtime/src/main/java/io/quarkus/smallrye/jwt/runtime/auth/DefaultJwtRolesMapper.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.smallrye.jwt.runtime.auth;
+
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import io.quarkus.arc.DefaultBean;
+
+@DefaultBean
+@ApplicationScoped
+public class DefaultJwtRolesMapper implements JwtRolesMapper {
```

Comment:
What the existing smallrye.jwt properties such as groups path for example can't do that this interface will help with ?

Reply:
At the moment I see three use cases:

1. the current implementation cannot handle JWTs like I showed in the example at the top. Sometimes tokens do not contain a groups field but contain role information in the form of other fields. In that example in the form of a boolean "admin" claim. With the `JwtRolesMapper`, I can write custom logic for my custom behavior to map the roles and then using the standard `@RolesAllowed` annotation.
2. It might well be that I do not want to use the group names as they are used in the JWT, which I might have to work with but have no control over. Sometimes the group names in the JWT are just not what they mean in my application. In that case, I can use the `JwtRolesMapper` to map the groups to roles that make sense in my context and thus make the code more readable.
3. I have a case where I have to handle tokens from different issuers, all using different groups. With the `JwtRolesMapper`, I can map their groups depending on the issuer of the token and unify it to something usable in my application. Again, I don't have power over those issuers, which is why I have to make it work in my application.